app/src/main/java/app/revanced/manager/patcher/worker/PatcherProgressManager.kt

```diff
@@ -9,117 +9,92 @@ import kotlinx.collections.immutable.persistentListOf
 import kotlinx.collections.immutable.toImmutableList
 import kotlinx.coroutines.flow.StateFlow
 
+enum class StepCategory(@StringRes val displayName: Int) {
+    PREPARING(R.string.patcher_step_group_preparing),
+    PATCHING(R.string.patcher_step_group_patching),
+    SAVING(R.string.patcher_step_group_saving)
+}
+
 enum class State {
-    WAITING, COMPLETED, FAILED
+    WAITING, RUNNING, FAILED, COMPLETED
 }
 
-class SubStep(
+data class Step(
     val name: String,
+    val category: StepCategory,
     val state: State = State.WAITING,
     val message: String? = null,
     val progress: StateFlow<Pair<Float, Float>?>? = null
 )
 
-class Step(
-    @StringRes val name: Int,
-    val subSteps: ImmutableList<SubStep>,
-    val state: State = State.WAITING
-)
-
 class PatcherProgressManager(
     context: Context,
     selectedPatches: List<String>,
     selectedApp: SelectedApp,
-    downloadProgress: StateFlow<Pair<Float, Float>?>
+    progress: StateFlow<Pair<Float, Float>?>
```

Comment:
Why was this renamed? the previous name seems better

Reply:
Because other steps might use the progress for other things in the future.

Reply:
The provided `StateFlow`is still tied the download step.

Reply:
Yeah but any step could be able to use it in the future

Reply:
Still feels like it makes more sense to preserve the name until another step needs it